src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForMultipleBuildingBlocks.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using MoBi.Assets;
+using MoBi.Core.Domain.Model;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Services;
+
+namespace MoBi.Presentation.Tasks.Interaction
+{
+   namespace MoBi.Presentation.Tasks.Interaction
+   {
+      public interface IInteractionTasksForMultipleBuildingBlocks
+      {
+         void RemoveBuildingBlocks(IReadOnlyList<IBuildingBlock> buildingBlocks);
+      }
+
+      public class InteractionTasksForMultipleBuildingBlocks : IInteractionTasksForMultipleBuildingBlocks
+      {
+         private readonly IInteractionTasksForChildren<Module, EventGroupBuildingBlock> _interactionTasksForApplicationMoleculeBuilder;
+         private readonly IInteractionTasksForChildren<Module, InitialConditionsBuildingBlock> _interactionTasksForInitialCondition;
+         private readonly IInteractionTasksForChildren<Module, MoleculeBuildingBlock> _interactionTasksForMolecule;
+         private readonly IInteractionTasksForChildren<Module, PassiveTransportBuildingBlock> _interactionTasksForPassiveTransport;
+         private readonly IInteractionTasksForChildren<Module, ParameterValuesBuildingBlock> _interactionTasksForParameterValues;
+         private readonly IInteractionTasksForChildren<Module, MoBiReactionBuildingBlock> _interactionTasksForMobiReaction;
+         private readonly IInteractionTasksForChildren<Module, MoBiSpatialStructure> _interactionTasksForMobiSpatialStructure;
+         private readonly IInteractionTasksForChildren<Module, ObserverBuildingBlock> _interactionTasksForObserver;
+         private readonly IInteractionTaskContext _interactionTaskContext;
+
+         public InteractionTasksForMultipleBuildingBlocks(
+            IInteractionTasksForChildren<Module, EventGroupBuildingBlock> interactionTasksForApplicationMoleculeBuilder,
+            IInteractionTasksForChildren<Module, InitialConditionsBuildingBlock> interactionTasksForInitialCondition,
+            IInteractionTasksForChildren<Module, MoleculeBuildingBlock> interactionTasksForMolecule,
+            IInteractionTasksForChildren<Module, PassiveTransportBuildingBlock> interactionTasksForPassiveTransport,
+            IInteractionTasksForChildren<Module, ParameterValuesBuildingBlock> interactionTasksForParameterValues,
+            IInteractionTasksForChildren<Module, MoBiReactionBuildingBlock> interactionTasksForMobiReaction,
+            IInteractionTasksForChildren<Module, MoBiSpatialStructure> interactionTasksForMobiSpatialStructure,
+            IInteractionTasksForChildren<Module, ObserverBuildingBlock> interactionTasksForObserver,
```

Comment:
These should be injected with the concrete class names. eg. `IInteractionTasksForMoleculeBuildingBlock`. I'm guessing you are getting the write classes, but the naming is off.

![image](https://github.com/user-attachments/assets/f3f79672-237b-4fb3-a29a-7ac7070ae9a0)


There are also some missing. For example `IndividualBuildingBlock` is not a child of `Module` so you don't have the interaction task for that. Same for `ExpressionBuildingBlock`

There should be a concrete type for all these

Reply:
You don't need to use the generic registration. You should just use the convention registration which means
IInteractionTasksForEventBuildingBlock => InteractionTasksForEventBuildingBlock. That's the easiest way to get the tasks you want.

That way you don't have to use the `Module` in the generic, and you can easily get the tasks for individual and expression because they are registered by convention.

Doesn't the multiselect context menu work with Individual BuildingBlocks and Expression Building Blocks?

Reply:
I made a mistake sayngi "These should be injected with the concrete class names. " This is not by class name, but by interface name. I just mean you should use the interface without the generics.